api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -124,4 +167,75 @@ private boolean isAllergyInactive(CodeableConcept status) {
 		        .anyMatch(c -> "inactive".equals(c.getCode()));
 	}
 	
+	private CodeableConcept getAllergySubstance(Allergen allergen) {
+		if (allergen.getCodedAllergen() == null) {
+			return null;
+		}
+		CodeableConcept allergySubstance = new CodeableConcept();
+		Coding code = new Coding(FhirConstants.ALLERGY_SUBSTANCE_VALUE_SET_URI, allergen.getCodedAllergen().getUuid(),
+		        allergen.getNonCodedAllergen());
+		allergySubstance.addCoding(code);
+		allergySubstance.setText("allergen.getNonCodedAllergen()");
+		
+		return allergySubstance;
+	}
+	
+	private CodeableConcept getManifestation(List<AllergyReaction> reactions) {
+		CodeableConcept manifestations = new CodeableConcept();
+		for (AllergyReaction reaction : reactions) {
+			Coding code = new Coding(FhirConstants.CLINICAL_FINDINGS_VALUE_SET_URI, reaction.getUuid(),
+			        reaction.getReactionNonCoded());
+			manifestations.addCoding(code);
+			manifestations.setText(reaction.getReactionNonCoded());
+		}
+		
+		return manifestations;
+	}
+	
+	private AllergyIntolerance.AllergyIntoleranceSeverity getFhirSeverity(Concept severityConcept) {
+		if (severityConcept == null) {
+			return null;
+		}
+		if (globalPropertyService.getGlobalProperty("allergy.concept.severity.mild") != null && globalPropertyService
```

Comment:
1. I'd prefer if this constant was referenced from a constant location, even if it ends up in `FhirConstants`.

2. Instead of calling `getGlobalProperty()` twice, call it once. This could be rewritten as `globalPropertyService.getGlobalProperty(FhirConstants.ALLERGY_MILD, "").equals(severityConcept.getUuid())`. This is because each call to `getGlobalProperty()` results in a query to the database.

Reply:
@ibacher i'm working on the tests, also i'll move those values to constant

Reply:
@ibacher without this check **globalPropertyService.getGlobalProperty("allergy.concept.severity.mild") != null**
i'm getting  **java.lang.NullPointerException**

Reply:
Right, sorry, I forgot this was our `GlobalProperty` service. Add a method to it like this:

```java
@Override
@Transactional(readOnly = true)
public String getGlobalProperty(String property, String defaultValue) throws APIException {
	String result = dao.getGlobalProperty(property);
	return result == null ? defaultValue : result;
}
```

and use that. Alternatively, use this pattern:

```java
String gp = globalPropertyService.getGlobalProperty(...);

if (gp != null && gp.equals(...)) {
	// ...
}
```

Reply:
Actually, change that first suggestion. Instead, on the `FhirGlobalProperty` interface add this:

```java
default String getGlobalProperty(String property, String defaultValue) throws APIException {
    String result = this.getGlobalProperty(property);
    return result == null ? defaultValue : result;
}
```

That way it works for any other implementations of `FhirGlobalProperty` without too much hassle.

Reply:
@ibacher adding change breaks this test  https://pastebin.com/6JG2P8b0 and I'm not sure how to fix it

here is the [branch](https://github.com/jecihjoy/openmrs-module-fhir2/tree/FM2-84-b) with these changes

How can I change [this](https://github.com/jecihjoy/openmrs-module-fhir2/blob/4450b4cc7d7127211e9c9b3b57f3182b9561d488/omod/src/test/java/org/openmrs/module/fhir2/web/servlet/BaseFhirResourceProviderTest.java#L99) to a normal java function?

Reply:
@jecihjoy I forgot about that. The equivalent is:

```java
servlet.setGlobalPropertyService(new FhirGlobalPropertyServiceImpl() {
	@Override
    @Transactional(readOnly = true)
    public String getGlobalProperty(String property) throws APIException {
        switch (property) {
			case FhirConstants.OPENMRS_FHIR_DEFAULT_PAGE_SIZE:
				return "10";
			case FhirConstants.OPENMRS_FHIR_MAXIMUM_PAGE_SIZE:
				return "100";
		}
    }
});
```

Reply:
I'm actually doing the mocking
eg  [this is](https://github.com/jecihjoy/openmrs-module-fhir2/blob/36501adfb9f8548f39e606f64e96ef73ef188d7e/api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImplTest.java#L290) where the tests starts failing if I remove the null check from the translator

The first element in the list, **MILD** works fine but the rest on the tests on severity won't pass